### PR TITLE
Use runtime-controller client interface

### PR DESCRIPTION
**Pull request by bboreham**
Instead of lower-level Kubernetes `clientSet`.

I think this is more idomatic for CAPI controllers.

We still need a `clientSet` to call `Drain()` from upstream.

### CODE REVIEW

File: pkg/apis/wksprovider/controller/wksctl/machine_controller.go [hunk 1]
```diff
@@ -300,15 +302,16 @@ func (a *MachineController) installNewBootstrapToken(ctx context.Context, ns str
 	if err != nil {
 		return nil, gerrors.Errorf("failed to create new bootstrap token %s/%s", ns, secret.ObjectMeta.Name)
 	}
-	s, err := a.clientSet.CoreV1().Secrets(ns).Create(ctx, secret, metav1.CreateOptions{})
+	err = a.client.Create(ctx, secret)
 	if err != nil {
 		return nil, gerrors.Errorf("failed to install new bootstrap token %s/%s", ns, secret.ObjectMeta.Name)
 	}
+	s := secret
```
Comment (twelho) [inline pkg/apis/wksprovider/controller/wksctl/machine_controller.go:309]: This assignment seems a bit redundant since `secret` is a pointer type.

Comment (bboreham) [reply pkg/apis/wksprovider/controller/wksctl/machine_controller.go:309]: Yeah it's to reduce diffs to make it easier to see what really changed. 
I can add another commit to squash the redundant stuff.

Comment (bboreham) [reply pkg/apis/wksprovider/controller/wksctl/machine_controller.go:309]: Actually I remember now; this code will change again when we fix #288 so I thought I'd just leave it for now.

File: pkg/apis/wksprovider/controller/wksctl/machine_controller.go [hunk 2]
```diff
@@ -300,15 +302,16 @@ func (a *MachineController) installNewBootstrapToken(ctx context.Context, ns str
 	if err != nil {
 		return nil, gerrors.Errorf("failed to create new bootstrap token %s/%s", ns, secret.ObjectMeta.Name)
 	}
-	s, err := a.clientSet.CoreV1().Secrets(ns).Create(ctx, secret, metav1.CreateOptions{})
+	err = a.client.Create(ctx, secret)
```
Comment (luxas) [inline pkg/apis/wksprovider/controller/wksctl/machine_controller.go:305]: Is it guaranteed that `secret` has the namespace populated correctly here?

File: pkg/apis/wksprovider/controller/wksctl/machine_controller.go [hunk 3]
```diff
@@ -203,7 +202,8 @@ func (a *MachineController) connectTo(ctx context.Context, c *existinginfrav1.Ex
 }
 
 func (a *MachineController) sshKey(ctx context.Context) ([]byte, error) {
-	secret, err := a.clientSet.CoreV1().Secrets(a.controllerNamespace).Get(ctx, controllerSecret, metav1.GetOptions{})
+	var secret v1.Secret
```
Comment (luxas) [inline pkg/apis/wksprovider/controller/wksctl/machine_controller.go:205]: shall we be consistent in passing secrets as pointers everywhere and make this `secret := &v1.Secret{}` instead?
Also, I'm not sure what happens when you pass the client a nil pointer tbh. Maybe it works.

File: pkg/apis/wksprovider/controller/wksctl/machine_controller.go [hunk 4]
```diff
@@ -27,7 +27,6 @@ import (
 	corev1 "k8s.io/api/core/v1"
 	v1 "k8s.io/api/core/v1"
```
Comment (luxas) [inline pkg/apis/wksprovider/controller/wksctl/machine_controller.go:28]: corev1 and v1 are imported twice, could you make this only corev1 please?

File: pkg/apis/wksprovider/controller/wksctl/machine_controller.go [hunk 5]
```diff
@@ -609,8 +612,8 @@ func (a *MachineController) getNodePlan(ctx context.Context, provider *existingi
 }
 
 func (a *MachineController) getAuthConfigMap(ctx context.Context) (*v1.ConfigMap, error) {
-	client := a.clientSet.CoreV1().ConfigMaps(a.controllerNamespace)
-	maps, err := client.List(ctx, metav1.ListOptions{})
+	var maps corev1.ConfigMapList
```
Comment (luxas) [inline pkg/apis/wksprovider/controller/wksctl/machine_controller.go:615]: same point as for the secret above

File: pkg/apis/wksprovider/controller/wksctl/machine_controller.go (hunk 3 above)
Comment (twelho) [reply pkg/apis/wksprovider/controller/wksctl/machine_controller.go]: I've updated these to use stack-allocated variables where possible as per our discussion. This eliminates the need to deal with potential `nil`s. Looking at the implementation of client it doesn't seem like passing `nil` would work anyways.

File: pkg/apis/wksprovider/controller/wksctl/machine_controller.go (hunk 2 above)
Comment (twelho) [reply pkg/apis/wksprovider/controller/wksctl/machine_controller.go:304]: Yes, it is always populated as `kube-system`.

**Pull request merged**